frontend/src/pages/notebookController/screens/admin/ServerStatus.tsx

```diff
@@ -44,23 +40,14 @@ const ServerStatus: React.FC<ServerStatusProps> = ({ data, username }) => {
   return (
     <Button
       variant="link"
-      isDisabled={deleting}
       isDanger
       isInline
       onClick={() => {
-        const notebookName = data.notebook?.metadata.name;
-        if (notebookName) {
-          setDeleting(true);
-          deleteNotebook(notebookNamespace, notebookName)
-            .then(() => data.forceRefresh())
-            .catch((e) => {
-              notification.error(`Error delete notebook ${notebookName}`, e.message);
-              setDeleting(false);
-            });
-        }
+        setNotebooksToStop(data.notebook ? [data.notebook] : []);
+        setServerStatuses([data]);
```

Comment:
If the data is already in `serverStatus` why are you setting back to back values... Also, `if (!data.notebook) {` above is already a check to make this `return` guaranteed to have a notebook

Reply:
the type seems broken here, I noticed the check above but it still gives me the warning.

Reply:
Ah crud... you are right 😕 Why is TS being so painful... that should have been vetted already. Okay, nvm my comment.